### PR TITLE
Add support for skybox transformation

### PR DESCRIPTION
**Pull request by Soulghost**
# Objective

- Fixes https://github.com/bevyengine/bevy/issues/14036

## Solution

- Add a view space transformation for the skybox

## Testing

- I have tested the newly added `transform` field using the `skybox` example.
```
diff --git a/examples/3d/skybox.rs b/examples/3d/skybox.rs
index beaf5b268..d16cbe988 100644
--- a/examples/3d/skybox.rs
+++ b/examples/3d/skybox.rs
@@ -81,6 +81,7 @@ fn setup(mut commands: Commands, asset_server: Res<AssetServer>) {
         Skybox {
             image: skybox_handle.clone(),
             brightness: 1000.0,
+            rotation: Quat::from_rotation_x(PI * -0.5),
         },
     ));
```
image

## Migration Guide
- Since we have added a new filed to the Skybox struct, users will need to include `..Default::default()` or some rotation value in their initialization code.

### CODE REVIEW

File: crates/bevy_core_pipeline/src/skybox/mod.rs [hunk 1]
```diff
@@ -90,6 +92,7 @@ pub struct Skybox {
     /// After applying this multiplier to the image samples, the resulting values should
     /// be in units of [cd/m^2](https://en.wikipedia.org/wiki/Candela_per_square_metre).
     pub brightness: f32,
+    pub transform: Option<Transform>,
```
Comment (hymm) [inline crates/bevy_core_pipeline/src/skybox/mod.rs:95]: Do the scale and translation do anything? If not this should probably be a quaternion.

Comment (Soulghost) [reply crates/bevy_core_pipeline/src/skybox/mod.rs]: It should only require Rotation. I have already adjusted the API to use quaternion, thank you.

**Comment by JMS55** [comment]
Imo the rotation should not be an Option, users can just use ..Skybox::default() if they don't want it.

Similarly, I don't think a pipeline specialization is worth it. I'd rather it just always be applied, with the default rotation being the identity.

**Comment by Soulghost** [comment]
> Imo the rotation should not be an Option, users can just use ..Skybox::default() if they don't want it.
> 
> Similarly, I don't think a pipeline specialization is worth it. I'd rather it just always be applied, with the default rotation being the identity.

I totally agree with your suggestions and have updated the code accordingly. Thanks!

**Review by Hexorg — APPROVED** [review]

File: crates/bevy_core_pipeline/src/skybox/mod.rs [hunk 2]
```diff
@@ -90,6 +92,17 @@ pub struct Skybox {
     /// After applying this multiplier to the image samples, the resulting values should
     /// be in units of [cd/m^2](https://en.wikipedia.org/wiki/Candela_per_square_metre).
     pub brightness: f32,
+    pub rotation: Quat,
```
Comment (alice-i-cecile) [inline crates/bevy_core_pipeline/src/skybox/mod.rs:95]: Docs would be nice here.

**Review by alice-i-cecile — APPROVED** [review]

File: crates/bevy_core_pipeline/src/skybox/mod.rs (hunk 2 above)
Comment (Soulghost) [reply crates/bevy_core_pipeline/src/skybox/mod.rs:99]: Documentation has been added to explain the rotation and its appropriate usage.

**Pull request merged**